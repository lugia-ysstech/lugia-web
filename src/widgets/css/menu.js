/**
 *
 * create by szfeng
 *
 * @flow
 */
import type { SizeType } from '../menu/item';
import CSSComponent, { css, StaticComponent } from '@lugia/theme-css-hoc';
import colorsFunc from './stateColor';
import { px2remcss } from '../css/units';
export const {
  themeColor,
  disableColor,
  blackColor,
  lightGreyColor,
  defaultColor,
  mediumGreyColor,
} = colorsFunc();
export const DefaultMenuItemHeight = 35;
export const LargeMenuItemHeight = 60;
export const BiggerMenuItemHeight = 40;
export const MenuItemHeight = 35;
export const DefaultHeight = 250;
export const DefaultWidth = 250;
export const ItemBackgroundColor = '#edf0fe';
export const SelectIcon = '\\e73e';
export const Height = 30;

export const getMenuItemHeight = (size: SizeType) => {
  return size === 'large'
    ? LargeMenuItemHeight
    : size === 'bigger'
    ? BiggerMenuItemHeight
    : DefaultMenuItemHeight;
};

export const SwitchIconContainer = StaticComponent({
  tag: 'span',
  className: 'SwitchIconContainer',
  css: css`
    position: absolute;
    right: ${px2remcss(12)};
    top: 50%;
    transform: translateY(-50%);
  `,
});

export const MenuContainer = CSSComponent({
  tag: 'ul',
  className: 'MenuContainer',
  normal: {
    selectNames: [['width'], ['background'], ['padding']],
    getCSS: (themeMeta, themeProps) => {
      const { menuItemHeight, length, autoHeight } = themeProps.propsConfig;
      let { height: themeHeight } = themeMeta;
      themeHeight = !themeHeight && themeHeight !== 0 ? DefaultHeight : themeHeight;
      const height = autoHeight ? menuItemHeight * length : themeHeight;
      return `height: ${px2remcss(height)};
      `;
    },
    defaultTheme: {
      background: {
        color: '#fff',
      },
      width: DefaultWidth,
    },
  },
  hover: {
    selectNames: [['background']],
  },
  css: css`
    outline: none;
    margin: 0;
    user-select: none;
    padding-left: 0;
    list-style: none;
    overflow: hidden;
    transition: all 0.3s;
  `,
  option: { hover: true },
});

const getIcon = checkedCSS => {
  return `
    ${
      checkedCSS !== 'mark'
        ? ''
        : `
    &::after {
      font-family: "sviconfont" !important;
      text-rendering: optimizeLegibility;
      content: "${SelectIcon}";
      color: transparent;
      display: inline-block;
      transform: scale(.83333333) rotate(0deg);
      zoom: 1;
      transition: all .2s ease;
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
      right: ${px2remcss(12)};
      font-weight: 700;
      font-size: ${px2remcss(16)};
      text-shadow: 0 0.1px 0, 0.1px 0 0, 0 -0.1px 0, -0.1px 0;
    }
    `
    }
  `;
};

export const ItemWrap = CSSComponent({
  tag: 'li',
  className: 'ItemWrap',
  normal: {
    selectNames: [
      ['height'],
      ['color'],
      ['font'],
      ['fontSize'],
      ['cursor'],
      ['border'],
      ['borderRadius'],
      ['padding'],
      ['background'],
      ['opacity'],
      ['boxShadow'],
    ],
    defaultTheme: {
      cursor: 'pointer',
    },
    getCSS: (themeMeta, themeProps) => {
      const { propsConfig: { menuItemHeight } = {} } = themeProps;

      return `
        height: ${px2remcss(menuItemHeight)};
        `;
    },
  },
  hover: {
    selectNames: [
      ['color'],
      ['font'],
      ['fontSize'],
      ['background'],
      ['opacity'],
      ['border'],
      ['borderRadius'],
      ['boxShadow'],
    ],
  },
  active: {
    selectNames: [
      ['color'],
      ['font'],
      ['fontSize'],
      ['background'],
      ['opacity'],
      ['border'],
      ['borderRadius'],
      ['boxShadow'],
    ],
  },
  disabled: {
    selectNames: [
      ['color'],
      ['font'],
      ['cursor'],
      ['borderRadius'],
      ['padding'],
      ['fontSize'],
      ['background'],
      ['opacity'],
    ],
    defaultTheme: {
      cursor: 'not-allowed',
      color: lightGreyColor,
      font: {
        fontWeight: 500,
      },
    },
  },
  css: css`
    box-sizing: border-box;
    position: relative;
    font-weight: 100;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    transition-property: background-color, border, borderRadius, opacity, boxShadow;
    transition-duration: 0.3s;
    font-size: ${px2remcss(14)};
    display: flex;
    flex-direction: column;

    & > i {
      vertical-align: middle;
    }
  `,
  option: { hover: true, active: true, disabled: true },
});
ItemWrap.displayName = 'ItemWrap';

export const DividerWrap = CSSComponent({
  tag: 'div',
  className: 'DividerWrap',
  normal: {
    selectNames: [['background']],
  },
  css: css`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    background: #666;
    height: ${px2remcss(1)};
  `,
});

export const TextContainer = CSSComponent({
  tag: 'div',
  className: 'TextContainer',
  normal: {
    selectNames: [['padding'], ['lineHeight']],
  },
  hover: {
    selectNames: [],
  },
  active: {
    selectNames: [],
  },
  css: css`
    padding: ${px2remcss(0)} ${px2remcss(8)};
    position: relative;
    display: flex;
    align-items: center;
    box-sizing: border-box;
    overflow: hidden;
    flex: 1;
    & i {
      vertical-align: middle;
    }
  `,
});

export const DesContainer = CSSComponent({
  tag: 'div',
  className: 'DesContainer',
  normal: {
    selectNames: [['color'], ['font'], ['fontSize'], ['padding'], ['lineHeight']],
  },
  hover: {
    selectNames: [['color'], ['font'], ['fontSize']],
  },
  active: {
    selectNames: [['color'], ['font'], ['fontSize']],
  },
  css: css`
    overflow: hidden;
    padding-left: ${px2remcss(10)};
    box-sizing: border-box;
    flex: 1;
    color: ${mediumGreyColor};
    font-weight: 100;
    display: flex;
    align-items: center;
    font-size: ${px2remcss(12)};
    transition: all 0.3s;
  `,
  option: { hover: true, active: true, disabled: true },
});

export const Text = StaticComponent({
  tag: 'span',
  className: 'Text',
  css: css`
    transition: all 0.3s;
  `,
});

export const SuffixElementWrap = StaticComponent({
  tag: 'div',
  className: 'SuffixElementWrap',
  css: css`
    transition: all 0.3s;
    position: absolute;
    right: 0;
    top: 50%;
    transform: translateY(-50%);
  `,
});
