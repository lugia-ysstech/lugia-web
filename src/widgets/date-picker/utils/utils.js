export const getformatSymbol = (value: string) => {
  const { length } = value;
  const symbolCont = [];
  const numberIndex = [];
  for (let i = 0; i < length; i++) {
    const numberValue = parseInt(value[i]);
    if (isNaN(numberValue)) {
      symbolCont.push(value[i]);
    } else {
      numberIndex.push(i);
    }
  }
  return {
    symbolCont,
    numberIndex,
  };
};
export const getTheme = (props: Object) => {
  const { getTheme } = props;
  const theme = getTheme();
  return { ...theme };
};
export function getNewProps(props: Object): Object {
  const filterParmas = { defaultVaule: 'defaultValue', value: 'value' };
  const newProps = {};
  for (const i in props) {
    if (i !== filterParmas[i]) {
      newProps[i] = props[i];
    }
  }
  return newProps;
}
export function tansValueFromStringToArray(props) {
  const filterParmas = { defaultValue: 'defaultValue', value: 'value', placeholder: 'placeholder' };
  const newProps = {};
  for (const i in props) {
    newProps[i] = props[i];
    if (i == filterParmas[i]) {
      newProps[i] = [props[i]];
    }
  }
  return newProps;
}
