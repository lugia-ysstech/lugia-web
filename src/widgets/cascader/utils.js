/**
 * create by szfeng
 *
 * @flow
 */
import { recurDataAndGetSelectedKeys } from '../menu/utils';

type CascaderProps = {
  getTheme: Function,
  offsetY: number,
  offsetX: number,
  action?: string,
  placeholder?: string,
  data?: Object[],
  onClick?: Function,
  onClear?: Function,
  onChange?: Function,
  separator?: string,
  value: string[],
  displayValue?: string[],
  disabled: boolean,
  displayField: string,
  valueField: string,
  showAllLevels?: boolean,
  allowClear?: boolean,
  menuWidth: number,
};
type CascaderState = {
  popupVisible: boolean,
  value: string[],
  expandedPath: string[],
  inputValue: string[],
  treeData: Array<Object>,
  selectedKeys: string[],
};

export function isHasValue(props: CascaderProps) {
  return 'value' in props;
}

export function isHasDefaultValue(props: CascaderProps) {
  return 'defaultValue' in props;
}

export function getLastIndex(array: Array<any>): number {
  return array.length - 1;
}

export function getInitExpandedPath(props: Object) {
  return getValue(props, null);
}

export function letStringToArray(value: string[]) {
  return Array.isArray(value) ? value : [value];
}

export function getValue(props: Object, state: Object | null): string[] {
  const { value = [], defaultValue = [] } = props;

  if (isHasValue(props)) {
    return value ? letStringToArray(value) : [];
  }
  if (!state) {
    return isHasDefaultValue(props) && defaultValue ? letStringToArray(defaultValue) : [];
  }
  return state.value;
}

export function isArrayLengthIsZero(value: string[]): boolean {
  return value.length === 0;
}

// 如果value是数字呢？
// split不Number的方法，如果传数字会报错
// 考虑容错
export function letValueSplitToArray(value: string[] = [], separator: string): string[] {
  if (Array.isArray(value)) {
    return isArrayLengthIsZero(value) ? [] : value[0].split(separator);
  }
  return value.split(separator);
}

export function getLastLevelValue(valueData: string[]): any {
  if (!valueData || isArrayLengthIsZero(valueData)) {
    return undefined;
  }
  const lastIndex = getLastIndex(valueData);
  return valueData[lastIndex];
}

export function isLeafPath(
  treeData: Array<Object>,
  pathArray: string[],
  valueField: string
): boolean {
  if (!treeData) {
    return false;
  }
  const key = getLastLevelValue(pathArray);
  if (!key) {
    return false;
  }
  const item = treeData && treeData.find(item => item[valueField] === key);
  if (!item) {
    return false;
  }
  return !!item.isLeaf;
}

export function mapTreeDataToGetDisplayValue(
  treeData: Array<Object>,
  filterKeys: string[],
  valueField: string,
  displayField: string
) {
  const displayValueData = [];
  if (!filterKeys || filterKeys.length === 0 || !treeData) {
    return displayValueData;
  }
  const exisitMap = filterKeys.reduce((exist: Object, key: string) => {
    exist[key] = true;
    return exist;
  }, {});

  treeData &&
    treeData.forEach(item => {
      const key = item[valueField];
      if (exisitMap[key]) {
        displayValueData.push(item[displayField]);
      }
    });

  return displayValueData;
}

export function getInitInputValue(props: CascaderProps) {
  const { displayValue } = props;
  return displayValue ? displayValue : [];
}

export function getInputValue(props: CascaderProps, state: CascaderState) {
  const {
    showAllLevels,
    separator = '|',
    data = [],
    valueField,
    displayField,
    treeData = [],
    displayValue = '',
  } = props;

  if (displayValue) {
    return displayValue;
  }

  const value = getValue(props, state);
  const filterValueData = getFilterValueData(data, value, separator);
  const displayValueData = mapTreeDataToGetDisplayValue(
    treeData,
    filterValueData,
    valueField,
    displayField
  );

  if (showAllLevels) {
    return isArrayLengthIsZero(displayValueData)
      ? displayValueData
      : [displayValueData.join(separator)];
  }

  const isLeaf = isLeafPath(treeData, filterValueData, valueField);
  return isLeaf ? [getLastLevelValue(displayValueData)] : state ? state.inputValue : '';
}

export function getFilterValueData(
  data: Array<Object>,
  value: string[],
  separator: string
): string[] {
  const valueData = letValueSplitToArray(value, separator);
  const filterData = [];
  recurDataAndGetSelectedKeys(data, valueData, filterData);
  return filterData;
}
