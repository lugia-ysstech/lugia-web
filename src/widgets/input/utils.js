import { ObjectUtils } from '@lugia/type-utils';
import { units } from '@lugia/css';
const { px2remcss } = units;

export const checkIsPercent = width => {
  return width && typeof width === 'string' && width.indexOf('%') !== -1;
};
export const getWidthCSS = width => {
  const theWidth = ObjectUtils.isNumber(width)
    ? px2remcss(width)
    : checkIsPercent(width)
    ? width
    : '100%';
  return `width: ${theWidth};`;
};
