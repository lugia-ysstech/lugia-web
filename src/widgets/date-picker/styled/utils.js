import colorsFunc from '../../css/stateColor';
import { px2remcss } from '../../css/units';
import { modeStyle } from '../utils/booleanUtils';
export const fontSize = 1.4;
export const em = px2remcss;

export const distance = {
  iconLeft: 10,
};
export const DateWrapperPadding = {
  top: 30,
  left: 20,
  bottom: 44,
};
export const borderRadius = 3;
const {
  hoverColor,
  normalColor,
  disableColor,
  spiritColor,
  lightGreyColor,
  borderDisableColor,
  borderSize,
  circleBorderRadius,
  defaultColor,
} = colorsFunc();
export const themeColor = {
  hoverColor,
  normalColor,
  disableColor,
  spiritColor,
  lightGreyColor,
  borderDisableColor,
  borderSize,
  circleBorderRadius,
  defaultColor,
};
export function getThemeProperty(props: Object) {
  const { hasTimeWrapBorder, hasItemNumber = 3 } = props;
  const { isRange, isTime } = modeStyle(props.mode);
  const normalWidth = isRange ? 420 : 300;
  const { width = normalWidth, color, backgroundColor } = props;
  const rangeWrapWidth = width >= 600 ? width : 600;
  const newWidth = isRange ? rangeWrapWidth / 2 : isTime ? width : width * 1;
  const TimeWrapWidth = isRange && hasTimeWrapBorder ? newWidth - 1 : newWidth;
  const TimeColWidth = TimeWrapWidth / hasItemNumber;
  const weekTitleWidth = (newWidth - DateWrapperPadding.left * 2 - 2) / 7;
  return {
    width: newWidth,
    TimeWrapWidth,
    color,
    backgroundColor,
    weekTitleWidth,
    rangeWrapWidth,
    TimeColWidth,
  };
}
export const getDateWrrap = () => {
  const { top, left } = DateWrapperPadding;
  const paddingStyle = ` ${em(top)}  ${em(left)} 0`;
  return {
    paddingStyle,
    left: em(left),
    top: em(top),
  };
};
