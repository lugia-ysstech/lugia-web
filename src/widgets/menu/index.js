import * as React from 'react';
import Menu from './menu';
import { getMenuItemHeight } from '../css/menu';
import ThemeHoc from '@lugia/theme-hoc';
import Widget from '../consts/index';
import Item from './item';
import { getTreeData } from './utils';

class ThrottleMenu extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.innerMenu = React.createRef();
  }

  render() {
    const { props } = this;
    const { children, getPartOfThemeConfig, __treeData__, separator = '|' } = props;
    const treeData =
      __treeData__ && __treeData__.length !== 0 ? __treeData__ : getTreeData(props, separator);
    const menuItemThemeConfig = getPartOfThemeConfig('MenuItem');
    const menuItemHeight = getMenuItemHeight(menuItemThemeConfig, props);
    return (
      <Menu ref={this.innerMenu} {...props} menuItemHeight={menuItemHeight} treeData={treeData}>
        {children}
      </Menu>
    );
  }
}

const MenuWrap = ThemeHoc(ThrottleMenu, Widget.Menu, {
  hover: true,
});

MenuWrap.MenuItem = Item;
export default MenuWrap;
