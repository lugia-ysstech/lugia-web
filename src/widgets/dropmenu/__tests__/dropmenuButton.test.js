/**
 * create by szfeng
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';

import DropMenu from '../';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Menu from '../../menu';
import DropMenuButton from '../dropmenuButton';
import renderer from 'react-test-renderer';
import Widget from '../../consts/index';
import Theme from '../../theme';
import { QueryInputPadding } from '../../common/QueryInputContainer';
import { adjustValue } from '../../utils';
import { delay } from '@lugia/react-test-utils';
import { MenuItemHeight, DefaultHeight, Height } from '../../css/menu';
const { MenuItem } = Menu;

Enzyme.configure({ adapter: new Adapter() });

const { expect: exp } = chai;

describe('DropMenuButton', () => {
  it('default', () => {
    const target = renderer.create(<DropMenuButton />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('divided={false}', () => {
    const target = renderer.create(<DropMenuButton divided={false} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('type=primary,divided={true}', () => {
    const target = renderer.create(<DropMenuButton type={'primary'} divided={true} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('type=basic', () => {
    const target = renderer.create(<DropMenuButton type={'basic'} divided={false} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  const items = [];
  for (let i = 0; i < 10; i++) {
    items.push(
      <MenuItem key={i} onClick={rest => {}}>
        {i}
      </MenuItem>
    );
  }

  const getTarget = (action, divided, type) =>
    class DropMenuDemo extends React.Component<any, any> {
      constructor(props) {
        super(props);
        this.state = {
          clickedState: false,
          visible: false,
          menu: <Menu onClick={(...rest) => {}}>{items}</Menu>,
        };
      }

      render() {
        const { menu } = this.state;
        const view = {
          [Widget.DropMenuButton]: { width: 180 },
          [Widget.DropMenu]: { width: 160 },
        };

        return (
          <Theme config={view}>
            <DropMenu
              onPopupVisibleChange={this.onPopupVisibleChange}
              menus={menu}
              action={action}
              hideAction={action}
            >
              <DropMenuButton type={type} divided={divided} onClick={this.onClick}>
                参数设置
              </DropMenuButton>
            </DropMenu>
          </Theme>
        );
      }

      onPopupVisibleChange = visible => {
        this.setState({ visible });
      };

      onClick = () => {
        this.setState({ clickedState: true });
      };
    };

  it('type: default, divided={true}; action = hover , click MenuItem && hover LeftButton', async () => {
    const Demo = getTarget('hover', true, 'customs');
    const cmp = mount(<Demo />);
    pullButtonMouseEnter(cmp);
    exp(cmp.state().visible).to.be.equal(true);
    await delay(100);
    cmp.update();

    pullButtonMouseLeave(cmp);
    await delay(100);
    exp(cmp.state().visible).to.be.equal(false);

    await delay(200);
    cmp.update();
    pullButtonMouseEnter(cmp);
    exp(cmp.state().visible).to.be.equal(true);
    await delay(200);
    cmp.update();
    clickMenuItem(cmp, 1);
    await delay(200);
    exp(cmp.state().visible).to.be.equal(false);

    hoverLeftButton(cmp);
    exp(cmp.state().visible).to.be.equal(false);
  });

  it('type: default, divided={true}, action = click , click MenuItem && click LeftButton', async () => {
    const Demo = getTarget('click', true, 'default');
    const cmp = mount(<Demo />);

    exp(cmp.state().visible).to.be.equal(false);

    pullButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    pullButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    pullButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);
    clickMenuItem(cmp, 1);

    cmp.update();
    await delay(300);

    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    clickLeftButton(cmp);
    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    exp(cmp.state().clickedState).to.be.equal(true);
  });

  it('type: primary, divided={false}; action = hover , click MenuItem && hover LeftButton', async () => {
    const Demo = getTarget('hover', false, 'primary');
    const cmp = mount(<Demo />);

    exp(cmp.state().visible).to.be.equal(false);

    NoDividedButtonMouseEnter(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    NoDividedButtonMouseLeave(cmp);

    cmp.update();
    await delay(300);

    exp(cmp.state().visible).to.be.equal(false);

    NoDividedButtonMouseEnter(cmp);

    cmp.update();
    await delay(300);

    exp(cmp.state().visible).to.be.equal(true);

    clickMenuItem(cmp, 1);

    exp(cmp.state().visible).to.be.equal(false);
  });

  it('type: primary, divided={false}, action = click , click MenuItem', async () => {
    const Demo = getTarget('click', false, 'primary');
    const cmp = mount(<Demo />);

    exp(cmp.state().visible).to.be.equal(false);
    NoDividedButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    NoDividedButtonClick(cmp);

    cmp.update();
    await delay(300);

    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    NoDividedButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    clickMenuItem(cmp, 1);
    exp(cmp.state().visible).to.be.equal(false);
  });

  it('type: basic, action = hover , hover MenuItem', async () => {
    const Demo = getTarget('hover', false, 'basic');
    const cmp = mount(<Demo />);
    exp(cmp.state().visible).to.be.equal(false);

    BasicButtonMouseEnter(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    BasicButtonMouseLeave(cmp);

    cmp.update();
    await delay(300);

    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    BasicButtonMouseEnter(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    clickMenuItem(cmp, 1);
    exp(cmp.state().visible).to.be.equal(false);
  });

  it('type: basic, action = click , click MenuItem', async () => {
    const Demo = getTarget('click', false, 'basic');
    const cmp = mount(<Demo />);
    exp(cmp.state().visible).to.be.equal(false);

    BasicButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    cmp.update();
    await delay(300);

    BasicButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(false);

    cmp.update();
    await delay(300);

    BasicButtonClick(cmp);
    exp(cmp.state().visible).to.be.equal(true);

    clickMenuItem(cmp, 1);
    exp(cmp.state().visible).to.be.equal(false);
  });

  function findPullButton(cmp: Object) {
    return cmp.find('DropMenuPullButton');
  }

  function pullButtonMouseEnter(cmp: Object) {
    return findPullButton(cmp).simulate('mouseEnter');
  }

  function pullButtonMouseLeave(cmp: Object) {
    return findPullButton(cmp).simulate('mouseLeave');
  }

  function pullButtonClick(cmp: Object) {
    return findPullButton(cmp).simulate('click');
  }

  function findMenuItem(cmp: Object) {
    return cmp
      .find(Widget.Menu)
      .at(0)
      .find(Widget.MenuItem);
  }

  function clickMenuItem(cmp: Object, index: number) {
    findMenuItem(cmp)
      .at(index)
      .simulate('click');
  }

  function findDevidedTextContainer(cmp: Object) {
    return cmp.find('DevidedTextContainer');
  }

  function hoverLeftButton(cmp: Object) {
    findDevidedTextContainer(cmp).simulate('mouseEnter');
  }

  function clickLeftButton(cmp: Object) {
    findDevidedTextContainer(cmp)
      .find('input')
      .at(0)
      .simulate('click');
  }

  function findNoDividedButton(cmp: Object) {
    return cmp.find(Widget.DropMenuButton).find('NoDividedContainer');
  }

  function NoDividedButtonMouseEnter(cmp) {
    findNoDividedButton(cmp).simulate('mouseEnter');
  }

  function NoDividedButtonMouseLeave(cmp) {
    findNoDividedButton(cmp).simulate('mouseLeave');
  }

  function NoDividedButtonClick(cmp) {
    findNoDividedButton(cmp).simulate('click');
  }

  function findBasicButton(cmp: Object) {
    return cmp.find(Widget.DropMenuButton).find('BasicWrap');
  }

  function BasicButtonMouseEnter(cmp) {
    findBasicButton(cmp).simulate('mouseEnter');
  }

  function BasicButtonMouseLeave(cmp) {
    findBasicButton(cmp).simulate('mouseLeave');
  }

  function BasicButtonClick(cmp) {
    findBasicButton(cmp).simulate('click');
  }
});
