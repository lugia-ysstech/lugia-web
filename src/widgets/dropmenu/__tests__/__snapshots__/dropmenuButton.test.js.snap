// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropMenuButton default 1`] = `
<div
  className="sc-gPEVay bSVfGR"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "backgroundColor": "#4d63ff",
      "borderBottomColor": "#4d63ff",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0.1rem",
      "borderLeftColor": "#4d63ff",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0.1rem",
      "borderRightColor": "#4d63ff",
      "borderRightStyle": "solid",
      "borderRightWidth": "0.1rem",
      "borderTopColor": "#4d63ff",
      "borderTopStyle": "solid",
      "borderTopWidth": "0.1rem",
      "color": "#ffffff",
      "height": "3.2rem",
    }
  }
>
  <input
    className="sc-iRbamj gSSfme"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    type="button"
  />
  <div
    className="sc-jlyJG gDwgBL"
  >
    <div
      className="sc-hzDkRC bCNFoS"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    />
    <i
      className="lugia-icon-direction_down   sc-VigVT bZGqSI"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
          "paddingBottom": "0rem",
          "paddingLeft": "0.6rem",
          "paddingRight": "0rem",
          "paddingTop": "0rem",
        }
      }
    />
  </div>
</div>
`;

exports[`DropMenuButton divided={false} 1`] = `
<div
  className="sc-gPEVay bSVfGR"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "backgroundColor": "#4d63ff",
      "borderBottomColor": "#4d63ff",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0.1rem",
      "borderLeftColor": "#4d63ff",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0.1rem",
      "borderRightColor": "#4d63ff",
      "borderRightStyle": "solid",
      "borderRightWidth": "0.1rem",
      "borderTopColor": "#4d63ff",
      "borderTopStyle": "solid",
      "borderTopWidth": "0.1rem",
      "color": "#ffffff",
      "height": "3.2rem",
    }
  }
>
  <input
    className="sc-iRbamj gSSfme"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    type="button"
  />
  <div
    className="sc-jlyJG gDwgBL"
  >
    <div
      className="sc-hzDkRC bCNFoS"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    />
    <i
      className="lugia-icon-direction_down   sc-VigVT bZGqSI"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
          "paddingBottom": "0rem",
          "paddingLeft": "0.6rem",
          "paddingRight": "0rem",
          "paddingTop": "0rem",
        }
      }
    />
  </div>
</div>
`;

exports[`DropMenuButton type=basic 1`] = `
<div
  className="sc-gPEVay bSVfGR"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(0,0,0,0)",
      "backgroundImage": "none",
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "color": "#4d63ff",
      "height": "3.2rem",
    }
  }
>
  <input
    className="sc-iRbamj gSSfme"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    type="button"
  />
  <div
    className="sc-jlyJG gDwgBL"
  >
    <div
      className="sc-hzDkRC bCNFoS"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    />
    <i
      className="lugia-icon-direction_down   sc-VigVT bZGqSI"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
          "paddingBottom": "0rem",
          "paddingLeft": "0.6rem",
          "paddingRight": "0rem",
          "paddingTop": "0rem",
        }
      }
    />
  </div>
</div>
`;

exports[`DropMenuButton type=primary,divided={true} 1`] = `
<div
  className="sc-csuQGl eauwPn"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "3.2rem",
    }
  }
  type="primary"
>
  <span
    checked={false}
    className="sc-fBuWsC hcHywv"
    disabled={false}
    type="primary"
  />
  <div
    className="sc-Rmtcm cTaZul"
  >
    <div
      checked={false}
      className="sc-bRBYWo dEFMCV"
      disabled={false}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "borderBottomColor": "#d0d3da",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0.1rem",
          "borderLeftColor": "#d0d3da",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0.1rem",
          "borderTopColor": "#d0d3da",
          "borderTopStyle": "solid",
          "borderTopWidth": "0.1rem",
          "color": "#363B40",
          "fontSize": "1.4rem",
        }
      }
      type="primary"
    >
      <input
        className="sc-iRbamj gSSfme"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        type="button"
      />
    </div>
    <span
      checked={false}
      className="sc-jhAzac fVFFSj"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "borderBottomColor": "#d0d3da",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0.1rem",
          "borderRightColor": "#d0d3da",
          "borderRightStyle": "solid",
          "borderRightWidth": "0.1rem",
          "borderTopColor": "#d0d3da",
          "borderTopStyle": "solid",
          "borderTopWidth": "0.1rem",
          "color": "#9EA4AE",
        }
      }
      type="primary"
    >
      <input
        className="sc-iRbamj gSSfme"
        disabled={false}
        onBlur={[Function]}
        type="button"
      />
      <i
        className="lugia-icon-direction_down   sc-VigVT jbKUDW"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "fontSize": "1.4rem",
          }
        }
      />
    </span>
  </div>
</div>
`;
